### PR TITLE
Refactor with split of log to CSV by log type

### PR DESCRIPTION
**Pull request by asmega**
# Context

- Partially related to https://digital.dclg.gov.uk/jira/browse/CLDC-2316
- Comprehending sales or lettings bulk upload CSV is enough, comprehending both simultaneously is rather challenging

# Changes

- Split out test helper class by log type ie lettings/sales

### CODE REVIEW

File: spec/support/bulk_upload/sales_log_to_csv.rb [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+  def row_prefix
+    [nil] * col_offset
+  end
+
+  def to_2022_csv_row
+    (row_prefix + to_2022_row).flatten.join(",") + line_ending
+  end
+
+  def default_2022_field_numbers
+    (1..134).to_a
```
Comment (SamSeed-Softwire) [inline spec/support/bulk_upload/sales_log_to_csv.rb]: Not relevant to this ticket but I think the number of fields for 2223 sales is either 125 or 128?

**Review by SamSeed-Softwire — APPROVED** [review]
Lgtm!

File: spec/support/bulk_upload/sales_log_to_csv.rb (hunk 1 above)
Comment (asmega) [reply spec/support/bulk_upload/sales_log_to_csv.rb]: ah good spot, you are correct it looks to be `125`, i will get that changed

Comment (asmega) [reply spec/support/bulk_upload/sales_log_to_csv.rb]: that's done now

**Pull request merged**